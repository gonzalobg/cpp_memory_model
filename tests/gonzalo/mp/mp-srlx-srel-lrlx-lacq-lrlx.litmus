C mp-srlx-srel-lrlx-lacq-lrlx
{ [x] = 0; [y] = 0;  }

P0 (atomic_int* x, int* y) {
  atomic_store_explicit(y, 1, memory_order_relaxed);
  atomic_store_explicit(x, 1, memory_order_release);
}

P1 (atomic_int* x, int* y) {
  int a = atomic_load_explicit(x, memory_order_relaxed);
  int c = atomic_load_explicit(x, memory_order_acquire);
  if (a == 1) {
    int b = atomic_load_explicit(y, memory_order_relaxed);
  }
}

P2 (atomic_int* x, int* y) {
  atomic_store_explicit(x, 2, memory_order_relaxed);
}

exists(1:a=1 /\ 1:b=0 /\ ~(1:c=1))