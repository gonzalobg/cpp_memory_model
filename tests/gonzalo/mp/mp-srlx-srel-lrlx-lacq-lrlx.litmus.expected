Test mp-srlx-srel-lrlx-lacq-lrlx Allowed
States 8
1:a=0; 1:b=0; 1:c=0;
1:a=0; 1:b=0; 1:c=1;
1:a=0; 1:b=0; 1:c=2;
1:a=1; 1:b=0; 1:c=2;
1:a=1; 1:b=1; 1:c=1;
1:a=1; 1:b=1; 1:c=2;
1:a=2; 1:b=0; 1:c=1;
1:a=2; 1:b=0; 1:c=2;
Ok
Witnesses
Positive: 1 Negative: 12
Condition exists (1:a=1 /\ 1:b=0 /\ not (1:c=1))
Observation mp-srlx-srel-lrlx-lacq-lrlx Sometimes 1 12
Hash=d91efcc59c21cecb451f30697c1c95e5

